### ISSUE TITLE
Namespaced Classes not support anymore

### ISSUE THREAD

**Issue by RobWu** [Shopify/identity_cache#191]
Looks like the master branch has now problems with namespaces classes.

Worked with older versions.

Error is created on here: https://github.com/Shopify/identity_cache/blob/master/lib/identity_cache/configuration_dsl.rb#L280 

Would suggest to use ``` .parameterize.underscore ``` instead of ``` .underscore ```

```
/Users/jom/.rvm/gems/ruby-2.1.2@uddannelsestesten-2015/bundler/gems/identity_cache-c314bfcf65eb/lib/identity_cache/configuration_dsl.rb:282:in `class_eval': /Users/jom/.rvm/gems/ruby-2.1.2@uddannelsestesten-2015/bundler/gems/identity_cache-c314bfcf65eb/lib/identity_cache/configuration_dsl.rb:288: syntax error, unexpected '/', expecting ';' or '\n' (SyntaxError)
          def expire_parent_cache_b56_survey/question
                                             ^
/Users/jom/.rvm/gems/ruby-2.1.2@uddannelsestesten-2015/bundler/gems/identity_cache-c314bfcf65eb/lib/identity_cache/configuration_dsl.rb:290: syntax error, unexpected keyword_end, expecting end-of-input
	from /Users/jom/.rvm/gems/ruby-2.1.2@uddannelsestesten-2015/bundler/gems/identity_cache-c314bfcf65eb/lib/identity_cache/configuration_dsl.rb:282:in `add_parent_expiry_hook'
	from /Users/jom/.rvm/gems/ruby-2.1.2@uddannelsestesten-2015/bundler/gems/identity_cache-c314bfcf65eb/lib/identity_cache/configuration_dsl.rb:221:in `build_recursive_association_cache'
	from /Users/jom/.rvm/gems/ruby-2.1.2@uddannelsestesten-2015/bundler/gems/identity_cache-c314bfcf65eb/lib/identity_cache/configuration_dsl.rb:105:in `cache_has_many'
...
```

**Comment by masterexploder** [comment]
This is present in the 0.2.3, and broke our apps as well. Would like to see this fixed soon, if possible, so we can keep taking new versions

**Comment by dylanahsmith** [comment]
@tjoyal looks like pull #184 caused this regression.

**Comment by lewisenator** [comment]
Changing lib/identity_cache/configuration_dsl.rb line 280 from
`after_action_name = "expire_parent_cache_#{self.name.underscore}"`
to
`after_action_name = "expire_parent_cache_#{self.name.underscore.gsub('/', '_')}"`
fixes the problem.

**Comment by dylanahsmith** [comment]
Pull #199 should also fix the issue, but what we could use is a regression test.

**Comment by deevis** [comment]
We're getting the same error with an upgrade to Rails 4.2.
Is this going to be addressed - I see there are two fixes now, neither pulled in though...

**Comment by midu** [comment]
+1

**Comment by RobWu** [comment]
Closed as pr #204 should fix it.
